booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,716 @@
+        var issues = new LinkedHashSet<String>();
+        switch (scenario) {
+          case REEFER, REEFER_TEMP_CHANGE -> {
+            if (!activeReeferNode.isObject()) {
+              issues.add("The scenario requires '%s' to have an active reefer".formatted(contextPath));
+            }
+          }
+          case REGULAR_NON_OPERATING_REEFER -> {
+            if (!nonOperatingReeferNode.asBoolean(false)) {
+              issues.add("The scenario requires '%s.isNonOperatingReefer' to be true".formatted(contextPath));
+            }
+          }
+          default -> {
+            if (!activeReeferNode.isMissingNode()) {
+              issues.add("The scenario requires '%s' to NOT have an active reefer".formatted(contextPath));
+            }
+            if (nonOperatingReeferNode.asBoolean(false)) {
+              issues.add("The scenario requires '%s.isNonOperatingReefer' to be omitted or false (depending on the container ISO code)".formatted(contextPath));
+            }
+          }
+        }
+        return issues;
+      }
+    ));
+
+    checks.add(JsonAttribute.allIndividualMatchesMustBeValid(
+      "[Scenario] Whether the cargo should be DG",
+      mav-> mav.path("requestedEquipments").all().path("commodities").all().path("outerPackaging").path("dangerousGoods").submitPath(),
+      (nodeToValidate, contextPath) -> {
+        var scenario = dspSupplier.get().scenarioType();
+        if (scenario == ScenarioType.DG) {
+          if (!nodeToValidate.isArray() || nodeToValidate.isEmpty()) {
+            return Set.of("The scenario requires '%s' to contain dangerous goods".formatted(contextPath));
+          }
+        } else {
+          if (!nodeToValidate.isMissingNode() || !nodeToValidate.isEmpty()) {
+            return Set.of("The scenario requires '%s' to NOT contain any dangerous goods".formatted(contextPath));
+          }
+        }
+        return Set.of();
+      }
+    ));
+  }
+
+  private static final JsonContentCheck VALIDATE_DOCUMENT_PARTIES = JsonAttribute.customValidator(
+    "Validate documentParties",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var documentParties = body.path("documentParties");
+      var partyFunctions = StreamSupport.stream(documentParties.spliterator(), false)
+        .map(p -> p.path("partyFunction"))
+        .filter(JsonNode::isTextual)
+        .map(n -> n.asText(""))
+        .collect(Collectors.toSet());
+
+      if (!partyFunctions.contains("SCO")) {
+        if (!body.path("serviceContractReference").isTextual()) {
+          issues.add("The 'SCO' party is mandatory when 'serviceContractReference' is absent");
+        }
+        if (!body.path("contractQuotationReference").isTextual()) {
+          issues.add("The 'SCO' party is mandatory when 'contractQuotationReference' is absent");
+        }
+      }
+      Set<String> partyShippers  = new HashSet<>(Arrays.asList("OS", "DDR"));
```

Comment:
Nit: The right hand side can be reduced to `Set.of("OS", "DDR");`

Reply:
done